src/MappedArrays.jl

```diff
@@ -62,9 +62,10 @@ parenttype{T,N,A,F,Finv}(::Type{MappedArray{T,N,A,F,Finv}}) = A
 
 function testvalue(data)
     if !isempty(data)
-        return first(data)
-    end
-    zero(eltype(data))
+        first(data)
+    else
+        zero(eltype(data))
+    end::eltype(data)
```

Comment:
Thinking about this a little I am sure how one would go about solving this dilemma. Is there even enough information at compile time? Since the collection can be empty (and need not be statically sized, so it is runtime information), probing actual elements can't be done during compile time, can it?